Feat/deploy docs (#295)
agno-agi/docs: deploy/introduction.mdx

## Overview

> Once you have built and configured your AgentOS locally, the next step is taking it to production. AgentOS exposes a FastAPI application under the hood, which means you are free to deploy it to any infrastructure you prefer such as Docker platforms, container services, serverless containers, Kubernetes, or a simple virtual machine.

Reviewer: ```suggestion
Once you have built and configured your AgentOS locally, the next step is taking it to production. AgentOS exposes a FastAPI application under the hood, which means you are free to deploy it to any infrastructure you prefer, including container services, serverless architectures, Kubernetes, or a simple virtual machine.
```

└── app/main.py # Houses your AgentOS app
```

Your agents, teams and workflows are automatically mounted and ready to serve.

## Adding a production ready Dockerfile

Here is a minimal Dockerfile for deploying AgentOS:

```docker
FROM agnohq/python:3.12

ENV PYTHONUNBUFFERED=1
PYTHONDONTWRITEBYTECODE=1
PIP_NO_CACHE_DIR=1
PIP_DISABLE_PIP_VERSION_CHECK=1
UV_CACHE_DIR=/tmp/uv-cache
UV_HTTP_TIMEOUT=120

WORKDIR /app

COPY requirements.txt pyproject.toml ./

RUN uv pip sync requirements.txt --system
 && uv cache clean

COPY . .

EXPOSE 8000

ENTRYPOINT ["/app/scripts/entrypoint.sh"]
CMD ["chill"]
```

This creates a clean, production-friendly container image.

## Testing locally

Build and run the container using the following commands from the root of your project:

Prerequisites:

- Docker installed
- Python 3.12 installed
- Agno installed

```bash
docker build -t my-agentos .
```

```bash
docker run -p 8000:8000 my-agentos
```

Visit `http://localhost:8000/docs` to see your AgentOS documentation. Here you can see the endpoints for your AgentOS.

## Deployment options

You can deploy AgentOS anywhere that supports Docker containers. The most common choices are listed below.

> ### AWS ECS

Reviewer: Maybe "Cloud" rather and then mention AWS in particular?

Author: If we are linking the templates then we kinda have to mention AWS

Recommended for production deployments. Our templates provide:

• multi architecture Docker builds  
• an ECR registry  
• a Fargate service for your container  
• a load balancer with HTTPS enabled  
• integration with RDS or any managed PostgreSQL service  
• logs, health checks, and VPC networking

### Railway

A simple and fast deployment method. Railway builds your Dockerfile, provides environment variables, and gives you a public domain with HTTPS automatically.

### Fly, Render, DigitalOcean

Great for smaller or simpler deployments. These platforms accept a Dockerfile, build it, and provide hosting with minimal configuration.

> ## Next steps

Reviewer: You can emulate the Cards layout used elsewhere
